build-aux/snap/snapcraft.yaml

```diff
@@ -28,174 +29,312 @@ license: GPL-3.0
 #
 # See the comments from jdstrand in
 # https://forum.snapcraft.io/t/5547/10
+
 parts:
-  snapd-deb:
+  dynamic-linker:
     plugin: nil
-    source: .
-    build-snaps: [go/1.18/stable]
-    # these packages are needed to call mkversion.sh in override-pull, all other
-    # dependencies are installed using apt-get build-dep
     build-packages:
-      - git
-      - dpkg-dev
-    after: [apparmor]
+      - gawk
+      - bison
+      - quilt
+      - python3-pyelftools
+    source: https://git.launchpad.net/ubuntu/+source/glibc
+    source-branch: ubuntu/jammy-updates
```

Comment:
anyone knows whether the actual git sha is written somewhere in the manifest?

Reply:
It is probably not. What I could do is to match the tag from the glibc package. And maybe git checkout to the right version.

I can also trick snapcraft by injecting an extended attribute to the installed file with the right version. Like `setfattr -n user.craft_parts.origin_stage_package -v libc6="$(git describe | sed 's,^.*/\([^/]*\)-[0-9]*-g[0-9a-f]*$,\1,')" "${CRAFT_PART_INSTALL}/usr/lib/${CRAFT_ARCH_TRIPLET_BUILD_FOR}/${DYNAMIC_LINKER}"`